File: easy_ecom/api/schemas/sales_agent.py
from __future__ import annotations

from typing import Any
from decimal import Decimal

from pydantic import BaseModel, Field, field_validator

from easy_ecom.api.schemas.commerce import LocationSummaryResponse, SalesOrderResponse


class ChannelIntegrationResponse(BaseModel):
    channel_id: str
    provider: str
    display_name: str
    status: str
    config_saved: bool
    webhook_key: str
    external_account_id: str
    phone_number_id: str
    phone_number: str
    verify_token_set: bool
    inbound_secret_set: bool
    access_token_set: bool
    webhook_verified_at: str | None
    last_webhook_post_at: str | None
    signature_validation_ok: bool | None
    graph_auth_ok: bool | None
    outbound_send_ok: bool | None
    openai_ready: bool
    openai_probe_ok: bool | None
    last_error_code: str | None
    last_error_message: str | None
    last_provider_status_code: int | None
    last_provider_response_excerpt: str | None
    last_diagnostic_at: str | None
    next_action: str
    default_location_id: str | None
    auto_send_enabled: bool
    agent_enabled: bool
    model_name: str
    persona_prompt: str
    config: dict[str, str]
    created_at: str
    updated_at: str
    last_inbound_at: str | None
    last_outbound_at: str | None


class ChannelIntegrationsResponse(BaseModel):
    items: list[ChannelIntegrationResponse]


class ChannelLocationsResponse(BaseModel):
    items: list[LocationSummaryResponse]


class WhatsAppMetaIntegrationRequest(BaseModel):
    display_name: str = "WhatsApp Sales Agent"
    external_account_id: str = ""
    phone_number_id: str = Field(min_length=1, max_length=128)
    phone_number: str = ""
    verify_token: str = ""
    access_token: str = ""
    app_secret: str = ""
    default_location_id: str | None = None
    auto_send_enabled: bool = False
    agent_enabled: bool = True
    model_name: str = "gpt-4o-mini"
    persona_prompt: str = ""


class WhatsAppMetaIntegrationResponse(BaseModel):
    channel: ChannelIntegrationResponse
    setup_verify_token: str | None = None


class ChannelDiagnosticsResponse(BaseModel):
    config_saved: bool
    verify_token_set: bool
    webhook_verified_at: str | None
    last_webhook_post_at: str | None
    signature_validation_ok: bool | None
    graph_auth_ok: bool | None
    outbound_send_ok: bool | None
    openai_ready: bool
    openai_probe_ok: bool | None
    last_error_code: str | None
    last_error_message: str | None
    last_provider_status_code: int | None
    last_provider_response_excerpt: str | None
    last_diagnostic_at: str | None
    next_action: str


class ChannelDiagnosticsEnvelopeResponse(BaseModel):
    diagnostics: ChannelDiagnosticsResponse
    provider_details: dict[str, Any]


class ChannelRunDiagnosticsResponse(BaseModel):
    channel: ChannelIntegrationResponse
    diagnostics: ChannelDiagnosticsResponse
    provider_details: dict[str, Any]


class ChannelSmokeRequest(BaseModel):
    recipient: str = Field(min_length=5, max_length=64)
    text: str = Field(default="EasyEcom smoke test. Reply path is working.", max_length=1000)


class ChannelSmokeResponse(BaseModel):
    ok: bool
    provider_event_id: str | None
    message: str
    diagnostics: ChannelDiagnosticsResponse
    provider_details: dict[str, Any]


class SalesAgentMentionResponse(BaseModel):
    mention_id: str
    product_id: str | None
    variant_id: str | None
    mention_role: str
    quantity: Decimal | None
    unit_price: Decimal | None
    min_price: Decimal | None
    available_to_sell: Decimal | None


class SalesAgentMessageResponse(BaseModel):
    message_id: str
    direction: str
    message_text: str
    content_summary: str
    occurred_at: str
    outbound_status: str
    provider_status: str
    mentions: list[SalesAgentMentionResponse]


class SalesAgentDraftResponse(BaseModel):
    draft_id: str
    conversation_id: str
    linked_sales_order_id: str | None
    status: str
    ai_draft_text: str
    edited_text: str
    final_text: str
    intent: str
    confidence: Decimal | None
    grounding: dict[str, object]
    reason_codes: list[str]
    approved_at: str | None
    sent_at: str | None
    failed_reason: str | None
    human_modified: bool


class SalesAgentConversationRowResponse(BaseModel):
    conversation_id: str
    channel_id: str
    customer_id: str | None
    customer_name: str
    customer_phone: str
    customer_email: str
    external_sender_id: str
    status: str
    customer_type: str
    behavior_tags: list[str]
    lifetime_spend: Decimal
    lifetime_order_count: int
    latest_intent: str
    latest_summary: str
    last_message_preview: str
    last_message_at: str | None
    latest_recommended_products_summary: str
    linked_draft_order_id: str | None
    linked_draft_order_status: str
    latest_draft_id: str | None
    latest_draft_status: str | None
    latest_trace: dict[str, object] = Field(default_factory=dict)
    linked_order: SalesOrderResponse | None


class SalesAgentConversationsResponse(BaseModel):
    items: list[SalesAgentConversationRowResponse]


class SalesAgentConversationDetailResponse(SalesAgentConversationRowResponse):
    messages: list[SalesAgentMessageResponse]
    latest_draft: SalesAgentDraftResponse | None


class SalesAgentHandoffRequest(BaseModel):
    notes: str = ""


class SalesAgentDraftApproveRequest(BaseModel):
    edited_text: str = ""


class SalesAgentDraftRejectRequest(BaseModel):
    reason: str = ""


class SalesAgentOrdersResponse(BaseModel):
    items: list[SalesOrderResponse]

File: frontend/types/sales-agent.ts
import type { SalesOrder } from '@/types/sales';

export type SalesAgentMention = {
  mention_id: string;
  product_id: string | null;
  variant_id: string | null;
  mention_role: string;
  quantity: string | null;
  unit_price: string | null;
  min_price: string | null;
  available_to_sell: string | null;
};

export type SalesAgentMessage = {
  message_id: string;
  direction: string;
  message_text: string;
  content_summary: string;
  occurred_at: string;
  outbound_status: string;
  provider_status: string;
  mentions: SalesAgentMention[];
};

export type SalesAgentDraft = {
  draft_id: string;
  conversation_id: string;
  linked_sales_order_id: string | null;
  status: string;
  ai_draft_text: string;
  edited_text: string;
  final_text: string;
  intent: string;
  confidence: string | null;
  grounding: Record<string, unknown>;
  reason_codes: string[];
  approved_at: string | null;
  sent_at: string | null;
  failed_reason: string | null;
  human_modified: boolean;
};

export type SalesAgentConversationRow = {
  conversation_id: string;
  channel_id: string;
  customer_id: string | null;
  customer_name: string;
  customer_phone: string;
  customer_email: string;
  external_sender_id: string;
  status: string;
  customer_type: string;
  behavior_tags: string[];
  lifetime_spend: string;
  lifetime_order_count: number;
  latest_intent: string;
  latest_summary: string;
  last_message_preview: string;
  last_message_at: string | null;
  latest_recommended_products_summary: string;
  linked_draft_order_id: string | null;
  linked_draft_order_status: string;
  latest_draft_id: string | null;
  latest_draft_status: string | null;
  latest_trace: Record<string, unknown>;
  linked_order: SalesOrder | null;
};

export type SalesAgentConversationDetail = SalesAgentConversationRow & {
  messages: SalesAgentMessage[];
  latest_draft: SalesAgentDraft | null;
};
